### PR TITLE
Fix URL routing for IOU

### PR DESCRIPTION
**Pull request by marcaaron**
cc @Julesssss 

### Fixed Issues
$ https://github.com/Expensify/Expensify.cash/issues/3886

### QA Steps
1. Navigate to e.cash
1. Click on FAB menu
1. Click on Split bill or request money
1. Verify no `%3AreportID`  is shown in the URL
1. Verify there are no other regressions 

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: src/ROUTES.js [hunk 1]
```diff
@@ -27,20 +31,23 @@ export default {
...
+    getIouSplitRoute: reportID => `${IOU_BILL}/${reportID}`,
+    getIOUSendRoute: reportID => `${IOU_SEND}/${reportID}`,
+    IOU_BILL_CURRENCY: `${IOU_BILL}/:reportID/currency`,
+    IOU_REQUEST_CURRENCY: `${IOU_REQUEST}/:reportID/currency`,
+    IOU_SEND_CURRENCY: `${IOU_SEND}/:reportID/currency`,
+    getIouRequestCurrencyRoute: reportID => `${IOU_REQUEST}/${reportID}/currency`,
+    getIouBillCurrencyRoute: reportID => `${IOU_BILL}/${reportID}/currency`,
+    getIouSendCurrencyRoute: reportID => `${IOU_SEND}/${reportID}/currency`,
+    IOU_DETAILS,
+    IOU_DETAILS_WITH_IOU_REPORT_ID: `${IOU_DETAILS}/:chatReportID/:iouReportID/`,
```
Comment (jasperhuangg) [inline src/ROUTES.js:50]: NAB: Have we defined a pattern for how we want to organize these routes? It seems like they're grouped by the pages they're related to. In addition, I personally think it's more clear if the getter for a route is placed right after the route itself.

**Review by jasperhuangg — APPROVED** [review]
looks good and tests well!

File: src/ROUTES.js (hunk 1 above)
Comment (marcaaron) [reply src/ROUTES.js:50]: >I personally think it's more clear if the getter for a route is placed right after the route itself.

I tried to reorganize things in a way that made sense to me, but definitely prioritizing the bug fix over the engineering efficiency of having things in a certain order. 😄

**Comment by roryabraham** [comment]
Unfortunately the deploy comments are not working right now. This was deployed to staging yesterday.

**Pull request merged**